pyproject.toml

```diff
@@ -26,24 +26,24 @@ classifiers=[
     "Programming Language :: Python :: 3.11",
 ]
 dependencies = [
-"numpy",
-"accelerate>=0.20.3",
-"transformers",
-"torch",
-"sentencepiece",
-"tokenizers>=0.13.3",
-"tqdm",
-"trl",
-"peft>=0.8.0",
-"datasets>=2.15.0",
-"fire",
-"simpleeval",
+"numpy>=1.26.4,<2.0",
+"accelerate>=0.20.3,<1.0",
+"transformers>=4.34.1,<5.0,!=4.38.2",
+"torch>=2.2.0,<3.0",
```

Comment:
May set to 2.4.0 current 2.3.1. I see API changes in minor release.

Reply:
since this one is above 1.0 and i think we would want to pull in new changes, unless something like dependabot is updating this i am more inclined to keep 3.0 as the upper limit